src/blpConnect.cpp

```diff
@@ -36,16 +41,45 @@ static void sessionFinalizer(SEXP session_) {
     }
 }
 
-// [[Rcpp::export]]
-SEXP blpConnect_Impl(const std::string host, const int port) {
+Session* blpConnectWithApp(const std::string host, const int port, const std::string app_name) {
     SessionOptions sessionOptions;
     sessionOptions.setServerHost(host.c_str());
     sessionOptions.setServerPort(port);
+    std::string app(app_name);
+    std::string authentication_string = APP_PREFIX + app;
+    sessionOptions.setAuthenticationOptions(authentication_string.c_str());
     Session* sp = new Session(sessionOptions);
 
     if (!sp->start()) {
-        Rcpp::stop("Failed to start session.\n");
+        Rcpp::stop("Failed to start session with bpipe app name.\n");
+    }
+
+    return sp;
+}
+Session* blpConnectNoApp(const std::string host, const int port) {
+    SessionOptions sessionOptions;
+    sessionOptions.setServerHost(host.c_str());
+    sessionOptions.setServerPort(port);
+    Session* sp = new Session(sessionOptions);
+
+    if (!sp->start()) {
+        Rcpp::stop("Failed to start session without an app.\n");
+    }
+
+    return sp;
+}
+
+// [[Rcpp::export]]
+SEXP blpConnect_Impl(const std::string host, const int port, SEXP app_name_) {
+    Session* sp = NULL;
+    if (app_name_ == NULL) {
```

Comment:
I think this should be `R_NilValue` rather than `NULL`. That seems to be what was killing my connection yesterday.